### PR TITLE
fix: we are sure about type if parent checked it

### PR DESCRIPTION
**Pull request by ChALkeR**
Type delta calculation here is needed for discriminator to be sure that we can compile.

Fixes: #176. 

This is just the first part of the fix that will work if `type: object` is specified on levels 1 and 2 or level 3.

An additional fix is needed to allow specifying `type: object` only once on level 1.

See also explanation in https://github.com/ExodusMovement/schemasafe/issues/176#issuecomment-1910259508

### CODE REVIEW

**Review by kklash — APPROVED** [review]

**Comment by mvayngrib** [comment]
is there a breaking unit test we can for this case?

**Comment by ChALkeR** [comment]
@mvayngrib done. #179 will improve it once this lands.

**Review by mvayngrib — APPROVED** [review]

File: test/regressions/177.js
Comment (ChALkeR) [reply test/regressions/177.js:24]: Ah. No, it should have been `{}`.
That doesn't affect anything but clarity though, as `{...[]}` is `{}`, arrays are spread by keys.

Comment (ChALkeR) [reply test/regressions/177.js:24]: Fixed!

**Review by mvayngrib — APPROVED** [review]

**Comment by ChALkeR** [comment]
This needs a second review per the new process.

**Comment by mvayngrib** [comment]
@kklash one more time please 😅. We're trying out a more strict review process. I'm already not a fan

**Review by kklash — APPROVED** [review]

**Review by kklash — APPROVED** [review]

**Comment by mvayngrib** [comment]
@ChALkeR this good to go?

**Pull request merged**
